bundles/org.openhab.core.persistence/src/main/java/org/openhab/core/persistence/extensions/RetimedHistoricItem.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.persistence.extensions;
+
+import java.time.ZonedDateTime;
+
+import org.openhab.core.persistence.HistoricItem;
+import org.openhab.core.types.State;
+
+/**
+ * This class wraps a HistoricItem to reset the timestamp.
+ *
+ * @author Florian Binder - Initial contribution
+ */
+public class RetimedHistoricItem implements HistoricItem {
```

Comment:
Please make this a `private static` inner class of `PersistenceExtensions`. We should not expose such "helper" classes to keep flexibility when doing refactoring in the future and avoid API breaking changes.

Reply:
Sure, did not know that.
Done.